LanguageFeatures/Patterns/variable_declaration_A05_t01.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The new rules are incorporated into the existing productions for
+/// declaring variables like so:
+///
+/// localVariableDeclaration ::=
+///   | metadata initializedVariableDeclaration ';' // Existing.
+///   | metadata patternVariableDeclaration ';' // New.
+///
+/// forLoopParts ::=
+///   | // Existing productions...
+///   | metadata ( 'final' | 'var' ) outerPattern 'in' expression // New.
+/// As with regular for-in loops, it is a compile-time error if the type of
+/// expression in a pattern-for-in loop is not assignable to Iterable<dynamic>.
+///
+/// @description Check that it is a compile-time error if the type of expression
+/// in a pattern-for-in loop is not assignable to Iterable<dynamic>.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_lib.dart";
+
+class Meta {
+  const Meta();
+}
+
+main() {
+  for (@Meta() final ((a1, b1) && r1) in (1, 2)) {
+//                                       ^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+  for (@Meta() var [a2, b2] in [1, 2]) {
```

Comment:
`[1, 2]` is assignable to `Iterable<dynamic>` no matter which type argument type inference chooses. So in that sense it doesn't really belong here. The error would complain about the pattern having a required type of `List<Something>`, and `int` not being assignable to that.

Maybe it's enough to make the '@description' a bit broader? Or put this into some other test?